common/go-utils/src/inv.rs

```diff
@@ -57,6 +48,7 @@ impl Inventory {
     pub fn resolve(&self, requirement: &Requirement) -> Option<&Artifact> {
         self.artifacts
             .iter()
+            .filter(|artifact| artifact.os == consts::OS && artifact.arch == consts::ARCH)
```

Comment:
is this filter necessary at this point since we're already filtered the available inventory to what the buildpack supports?

Reply:
It's not necessary, but I included a few notes about why I put it here in this PR up in the PR description:

> Finally, `Inventory::resolve` now filters artifacts based on the current OS and ARCH prior to resolving artifacts that satisfy the version string requirements. This is based on the assumption that the target platform/runtime will always be the same as the build platform/environment executing the buildpack.
> 
> While adding support for updating and parsing artifacts for other OS and ARCH combinations than `linux`/`x86_64` is outside the scope of this PR (particularly so the `inventory.toml` field changes can be reviewed on their own, without adding more `artifact` entries), this seems to be the appropriate time and place to introduce this constraint - ensuring the new fields are also being utilized in the code that consumes our inventory.

Then this PR (still a draft) will be responsible writing the additional artifacts to the inventory.toml (with the consumer of that file e.g., `Inventory::resolve` already supporting it)

Reply:
Much appreciated! Already added it here https://github.com/heroku/buildpacks-go/pull/229/commits/3c70b3c03c378c5cf54131e6bc9762fd7c91fb61